src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -5234,6 +5216,12 @@ public override IOperation VisitInstanceReference(IInstanceReferenceOperation op
                 {
                     return OperationCloner.CloneOperation(_currentImplicitInstance.ImplicitInstance);
                 }
+                else if (operation.Type.IsAnonymousType &&
+                    (object)_currentImplicitInstance.AnonymousType == operation.Type)
+                {
+                    return new InstanceReferenceExpression(operation.ReferenceKind, semanticModel: null, operation.Syntax, operation.Type, 
```

Comment:
>return new InstanceReferenceExpression(operation.ReferenceKind, semanticModel: null, operation.Syntax, operation.Type,  [](start = 20, length = 119)

Is this code reachable? Can you point me to a unit-test that hits it?

Reply:
Newly added test `WithFlow_07` hits this for `IInstanceReferenceOperation (ReferenceKind: ImplicitReceiver) (OperationKind.InstanceReference, Type: <anonymous type: I As System.Int32>, IsImplicit) (Syntax: 'New With { .I = 0 }')`

---
In reply to: [197280509](https://github.com/dotnet/roslyn/pull/28005#discussion_r197280509) [](ancestors = 197280509)

Reply:
> Newly added test WithFlow_07 hits this for IInstanceReferenceOperation (ReferenceKind: ImplicitReceiver) (OperationKind.InstanceReference, Type: <anonymous type: I As System.Int32>, IsImplicit) (Syntax: 'New With { .I = 0 }')

I see, I think it would be better to handle the implicit receivers explicitly in VisitAnonymousObjectCreation instead of trying to handle them here, potentially handling cases that we do not intend to handle. 

---
In reply to: [197285755](https://github.com/dotnet/roslyn/pull/28005#discussion_r197285755) [](ancestors = 197285755,197280509)

Reply:
Sure, refactored the code.

---
In reply to: [197290048](https://github.com/dotnet/roslyn/pull/28005#discussion_r197290048) [](ancestors = 197290048,197285755,197280509)